### ISSUE TITLE
[Bug]: dear teachers, when i change the gelu on spu, the result always similar very much, even when i change the gelu at a worse format

### ISSUE THREAD

**Issue by zhangwaer** [secretflow/spu#758]
### Issue Type

Others

### Modules Involved

Others

### Have you reproduced the bug with SPU HEAD?

Yes

### Have you searched existing issues?

Yes

### SPU Version

spu0.9.1b0

### OS Platform and Distribution

ubuntu18.04

### Python Version

3.10.14

### Compiler Version

GCC11.2.1

### Current Behavior?

[Bug]: dear teachers, when i change the gelu① of gpt2_version on spu, the result always similar very much, even when i change the gelu at a worse format gelu②. could you please give me some advice on how to change the gelu. Thank you very much!!!

gelu①
def hack_gelu(
    x: torch.Tensor,
    axis: Optional[Union[int, Tuple[int, ...]]] = -1,
    where: Optional[torch.Tensor] = None,
    initial: Optional[torch.Tensor] = None,
) -> torch.Tensor:
    b0 = x < -4.0
    b1 = x < -1.95
    b2 = x > 3.0
    b3 = b1 ^ b2 ^ True  # x in [-1.95, 3.0]
    b4 = b0 ^ b1  # x in [-4, -1.95]
    # seg1 = a[3] * x^3 + a[2] * x^2 + a[1] * x + a[0]
    # seg2 = b[6] * x^6 + b[4] * x^4 + b[2] * x^2 + b[1] * x + b[0]
    a_coeffs = torch.tensor(
        [
            -0.5054031199708174,
            -0.42226581151983866,
            -0.11807612951181953,
            -0.011034134030615728,
        ],
        dtype=x.dtype,
        device=x.device
    )
    b_coeffs = torch.tensor(
        [
            0.008526321541038084,
            0.5,
            0.3603292692789629,
            0.0,
            -0.037688200365904236,
            0.0,
            0.0018067462606141187,
        ],
        dtype=x.dtype,
        device=x.device
    )
    x2 = torch.square(x)
    x3 = x * x2
    x4 = torch.square(x2)
    x6 = torch.square(x3)
    seg1 = a_coeffs[3] * x3 + a_coeffs[2] * x2 + a_coeffs[1] * x + a_coeffs[0]
    seg2 = (
        b_coeffs[6] * x6
        + b_coeffs[4] * x4
        + b_coeffs[2] * x2
        + b_coeffs[1] * x
        + b_coeffs[0]
    )
    ret = b2 * x + b4 * seg1 + b3 * seg2
    return ret

@contextmanager
def hack_gelu_context(msg: str, enabled: bool = False):
    if not enabled:
        yield
        return
    # hijack some target functions
    raw_gelu = F.gelu
    F.gelu = hack_gelu
    print(msg)  # Optionally, print or log the message
    yield
    # recover back
    F.gelu = raw_gelu

gelu②
def hack_gelu(
    x: torch.Tensor,
    axis: Optional[Union[int, Tuple[int, ...]]] = -1,
    where: Optional[torch.Tensor] = None,
    initial: Optional[torch.Tensor] = None,
) -> torch.Tensor:
    b0 = x < -4.0
    b1 = x < -1.95
    b2 = x > 3.0
    x2 = torch.square(x)
    x3 = x * x2
    x4 = torch.square(x2)
    x6 = torch.square(x3)
    seg1 =  x3 +  x2 +  x 
    ret = b2 * x + b4 * seg1
    return ret
@contextmanager
def hack_gelu_context(msg: str, enabled: bool = False):
    if not enabled:
        yield
        return
    # hijack some target functions
    raw_gelu = F.gelu
    F.gelu = hack_gelu
    print(msg)  # Optionally, print or log the message
    yield
    # recover back
    F.gelu = raw_gelu

### Standalone code to reproduce the issue

```Python
[Bug]
```

**Comment by tpppppub** [comment]
I am not familiar with GPT-2 model details but I don't find anywhere the huggingface gpt-2 model calling F.gelu. (https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt2/modeling_gpt2.py). Any reason to replace F.gelu if they are never used?

**Comment by zhangwaer** [comment]
> I am not familiar with GPT-2 model details but I don't find anywhere the huggingface gpt-2 model calling F.gelu. (https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt2/modeling_gpt2.py). Any reason to replace F.gelu if they are never used?
thank you very much, i understand your teaching, thank you very much!! i will replace the actiation correctly.
